src/Game/AccountManagement.java

```diff
@@ -114,6 +123,222 @@ public static void register(String user, String pass) {
       }
     }
   }
+  
+  public static void forgotPass(String user) {
+	  Client.setLoginMessage("Connecting to server", "Please wait...");
+
+		try {
+			int port = Replay.connection_port;
+	        StreamUtil.initializeStream(Client.server_address, port);
+	        StreamUtil.setStreamMaxRetries(Client.maxRetries);
+
+	        Client.session_id = 0; // TODO: should read int to get session ID here, triggered by TCP handshake
+	        
+	        StreamUtil.newPacket(4);
+	        Object buffer = StreamUtil.getStreamBuffer();
+	        StreamUtil.putLongTo(buffer, Util.username2hash(user));
+	        StreamUtil.flushPacket();
+	        
+			StreamUtil.readShort(); // Unknown what data this contained, client doesn't use it.
+			int response = StreamUtil.readStream();
+			Logger.Game("Getpq response: " + response);
+			if (response == 0) {
+				Client.setLoginMessage("", "Sorry, the recovery questions for this user have not been set");
+				return;
+			}
+
+			String question;
+			int idx;
+			for (int var12 = 0; var12 < 5; ++var12) {
+				int length = StreamUtil.readStream();
+                byte[] arr = new byte[5000];
+                StreamUtil.readBytes(arr, length);
+                question = new String(arr, 0, length);
+                
+                if (question.startsWith("~:")) {

```

Comment:
I don't see this logic in RSC175

Reply:
That is sort of for compatibility, like for OpenRSC since recovery questions code was taken from 127, where the question sent starts with ~:

Reply:
OK. I can see that in 127. I also see your note above the entire function that this block is from 127.

Still I think that insert on line 157 `\n                     // This section is for parsing questions set in RSC127`

Reply:
placed